openlibrary/plugins/worksearch/schemes/works.py

```diff
@@ -39,173 +39,168 @@
+            'rating': 'ratings_sortable desc',
+            'rating asc': 'ratings_sortable asc',
+            'rating desc': 'ratings_sortable desc',
+            'readinglog': 'readinglog_count desc',
+            'want_to_read': 'want_to_read_count desc',
+            'currently_reading': 'currently_reading_count desc',
+            'already_read': 'already_read_count desc',
+            'title': 'title_sort asc',
+            'scans': 'ia_count desc',
+            # Classifications
+            'lcc_sort': 'lcc_sort asc',
+            'lcc_sort asc': 'lcc_sort asc',
+            'lcc_sort desc': 'lcc_sort desc',
+            'ddc_sort': 'ddc_sort asc',
+            'ddc_sort asc': 'ddc_sort asc',
+            'ddc_sort desc': 'ddc_sort desc',
+            # Ebook access
+            'ebook_access': 'ebook_access desc',
+            'ebook_access asc': 'ebook_access asc',
+            'ebook_access desc': 'ebook_access desc',
+            # Open Syllabus Project
+            'osp_count': 'osp_count desc',
+            'osp_count asc': 'osp_count asc',
+            'osp_count desc': 'osp_count desc',
+            # Key
+            'key': 'key asc',
+            'key asc': 'key asc',
+            'key desc': 'key desc',
+            # Random
+            'random': 'random_1 asc',
+            'random asc': 'random_1 asc',
+            'random desc': 'random_1 desc',
+            'random.hourly': lambda: f'random_{datetime.now():%Y%m%dT%H} asc',
+            'random.daily': lambda: f'random_{datetime.now():%Y%m%d} asc',
+        }
+        self.default_fetched_fields = {
+            'key',
+            'author_name',
+            'author_key',
+            'title',
+            'subtitle',
+            'edition_count',
+            'ia',
             'has_fulltext',
-            'false',
-        ): lambda: f'ebook_access:[* TO {get_fulltext_min()}}}',
-    }
+            'first_publish_year',
+            'cover_i',
+            'cover_edition_key',
+            'public_scan_b',
+            'lending_edition_s',
+            'lending_identifier_s',
+            'language',
+            'ia_collection_s',
+            # FIXME: These should be fetched from book_providers, but can't cause circular
+            # dep
+            'id_project_gutenberg',
+            'id_project_runeberg',
+            'id_librivox',
+            'id_standard_ebooks',
+            'id_openstax',
+            'id_cita_press',
+            'id_wikisource',
+        }
+        self.facet_rewrites = {
+            ('public_scan', 'true'): 'ebook_access:public',
+            ('public_scan', 'false'): '-ebook_access:public',
+            ('print_disabled', 'true'): 'ebook_access:printdisabled',
+            ('print_disabled', 'false'): '-ebook_access:printdisabled',
+            ('has_fulltext', 'true'): f'ebook_access:[{get_fulltext_min()} TO *]',
+            ('has_fulltext', 'false'): f'ebook_access:[* TO {get_fulltext_min()}]',
```

Comment:
These will need their lambdas back, because the value needs to be computed on a per-user basis.